src/peft/tuners/tuners_utils.py

```diff
@@ -523,3 +528,37 @@ def inspect_matched_modules(tuner: BaseTuner, adapter_name: str = "default") ->
         else:
             module_dict["unmatched"].append(key)
     return module_dict
+
+
+def _maybe_include_all_linear_layers(peft_config: PeftConfig, model: nn.Module) -> PeftConfig:
+    """
+    Helper function to update `target_modules` to all linear/Conv1D layers if provided as 'ALL'. Adapted from the QLoRA
+    repository: https://github.com/artidoro/qlora/blob/main/qlora.py
+    """
+    if peft_config.target_modules == INCLUDE_ALL_LAYERS_SHORTHAND:
+        is_loaded_in_8bit = getattr(model, "is_loaded_in_8bit", False)
+        is_loaded_in_4bit = getattr(model, "is_loaded_in_4bit", False)
+        # match with a list of linear layer classes. this is needed as sometimes you can
+        # have a mix Eg. T5 with 8bit has instances of torch.nn.Linear and bnb.nn.Linear8bitLt
+        linear_classes = [torch.nn.Linear, Conv1D]
+        if is_bnb_available():
+            import bitsandbytes as bnb
+
+            linear_classes = [bnb.nn.Linear8bitLt] + linear_classes if is_loaded_in_8bit else linear_classes
+            linear_classes = [bnb.nn.Linear4bit] + linear_classes if is_loaded_in_4bit else linear_classes
+
+        linear_module_names = set()
+        for name, module in model.named_modules():
+            # match with all linear classes.
+            for cls in linear_classes:
+                if isinstance(module, cls):
```

Comment:
You can pass multiple classes at once to `isinstance` instead of looping.

Reply:
Fixed! Actually didn't know about this :p